blueprints/materials/concrete.py

```diff
@@ -1 +1,547 @@
+
+        Returns
+        -------
+        MPA
+            Example: 32836 (for C30/37)
+        """
+        if self._e_c:
+            return self._e_c
+        return self.e_cm
+
+    @property
+    def concrete_class(self) -> ConcreteStrengthClass:
+        """Concrete class of the ConcreteMaterial object.
+
+        Returns
+        -------
+        ConcreteStrengthClass
+            Example: "ConcreteStrengthClass.C30_37"
+        """
+        return self._concrete_class
+
+    @property
+    def material_factor(self) -> DIMENSIONLESS:
+        """Partial safety factor (yc) for concrete according to NEN-EN 1992-1-1 art.2.4.2.4[-].
+
+        Returns
+        -------
+        DIMENSIONLESS
+            Example: 1.5
+        """
+        return self._material_factor
+
+    @property
+    def thermal_coefficient(self) -> DIMENSIONLESS:
+        """Thermal coefficient of the material [-].
+
+        Returns
+        -------
+        DIMENSIONLESS
+            Example: 1e-5
+        """
+        return self._thermal_coefficient
+
+    @property
+    def f_ck(self) -> MPA:
+        """Characteristic compressive cylinder strength of concrete at 28 days [MPa].
+
+        Returns
+        -------
+        MPA
+            Example: 30 (for C30/37)
+        """
+        value = self._concrete_class.value
+        match = re.search(pattern=r"C(\d+)/", string=value)
+        assert match
+        return int(match.group(1))
+
+    @property
+    def f_ck_cube(self) -> MPA:
+        """Characteristic compressive cubic strength of concrete_checks at 28 days [MPa].
+
+        Returns
+        -------
+        MPA
+            Example: 37 (for C30/37)
+        """
+        value = self._concrete_class.value
+        match = re.search(pattern=r"/(\d+)", string=value)
+        assert match
+        return int(match.group(1))
+
+    @property
+    def f_cd(self) -> MPA:
+        """Design compressive cylinder strength of concrete_checks at 28 days (NEN-EN 1992-1-1 art.3.1.6 (1)) [MPa].
+
+        Returns
+        -------
+        MPA
+            Example: 20 (for C30/37)
+        """
+        return self.f_ck / self.material_factor
+
+    @property
+    def f_cm(self) -> MPA:
+        """Mean value of concrete cylinder compressive strength [MPa].
+
+        Returns
+        -------
+        MPA
+            Example: 38 (for C30/37)
+        """
+        return self.f_ck + 8
+
+    @property
+    def f_cm_cube(self) -> MPA:
+        """Mean value of concrete_checks cubic compressive strength [MPa].
+
+        Returns
+        -------
+        MPA
+            Example: 45 (for C30/37)
+        """
+        return self.f_ck_cube + 8
+
+    @property
+    def f_ctm(self) -> MPA:
+        """Mean value of axial tensile strength of concrete [MPa].
+
+        Returns
+        -------
+        MPA
+            Example: 2,9 (for C30/37)
+        """
+        if self.f_ck <= 50:
+            return 0.30 * self.f_ck ** (2 / 3)
```

Comment:
W should implement these formulas as 'subformulas' and make them 'standalone' available --> not need for now, it works.